src/components/instantlaunches/admin/InstantLaunchMapping.js

```diff
@@ -0,0 +1,452 @@
+
+                <MenuItem
+                    value="infoType"
+                    key="infoType"
+                    id={buildID(
+                        baseID,
+                        ids.PATTERN,
+                        ids.KIND,
+                        ids.MENU,
+                        ids.INFO_TYPE
+                    )}
+                >
+                    {t("infoType")}
+                </MenuItem>
+            </TextField>
+
+            <TextField
+                id={buildID(baseID, ids.PATTERN)}
+                name="pattern"
+                label={t("pattern")}
+                value={formik.values.pattern}
+                onChange={formik.handleChange}
+                className={classes.flexItem}
+                error={formik.touched.pattern && formik.errors.pattern}
+                helperText={formik.touched.pattern && formik.errors.pattern}
+            />
+
+            <TextField
+                id={buildID(baseID, ids.BASE, ids.MENU)}
+                value={formik.values.instantLaunch}
+                onChange={formik.handleChange}
+                name="instantLaunch"
+                label={t("instantLaunch")}
+                className={classes.flexItem}
+                select
+            >
+                {instantlaunches.map((il, index) => {
+                    return (
+                        <MenuItem
+                            value={index}
+                            key={il.id}
+                            id={buildID(baseID, ids.BASE, ids.MENU, index)}
+                        >
+                            {il.quick_launch_name}
+                        </MenuItem>
+                    );
+                })}
+            </TextField>
+
+            <Button
+                type="submit"
+                variant="contained"
+                color="primary"
+                startIcon={<AddIcon />}
+                className={`${classes.submitButton} ${classes.flexItem}`}
+                id={buildID(baseID, ids.ADD, ids.BUTTON)}
+            >
+                {t("common:add")}
+            </Button>
+        </form>
+    );
+};
+
+const InstantLaunchMappingEditor = ({ showErrorAnnouncer }) => {
+    const baseID = buildID(ids.BASE, ids.MAPPING, ids.EDITOR);
+
+    const classes = useStyles();
+
+    const { t } = useTranslation(["instantlaunches", "common"]);
+
+    const defaultsMapping = useQuery(
+        DEFAULTS_MAPPING_QUERY_KEY,
+        getDefaultsMapping
+    );
+
+    const instantlaunches = useQuery(
+        ALL_INSTANT_LAUNCHES_KEY,
+        listFullInstantLaunches
+    );
+
+    const allQL = useQuery(
+        LIST_PUBLIC_QUICK_LAUNCHES_KEY,
+        getPublicQuicklaunches
+    );
+
+    const handleDelete = async (index) => {
+        const newEntries = Object.entries(defaultsMapping.data.mapping);
+        newEntries.splice(index, 1);
+        const newObj = Object.fromEntries(newEntries);
+        return await updateDefaultsMapping(newObj);
+    };
```

Comment:
Another reason to change this: you can't depend on on the Object.entries() function to return the results in the same order every time and we were passively assuming that would be the case here. 

The code I wrote is a little different, since `delete` gives me the willies for some reason:

```
    const handleDelete = async (mappingName) => {
        const {
            [mappingName]: _,
            ...newMapping
        } = defaultsMapping.data.mapping;
        return await updateDefaultsMapping(newMapping);
    };
```